Algorithms/Problems/StableMarriage/GaleShapley.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Algorithms.Problems.StableMarriage
+{
+    public static class GaleShapley
+    {
+        /// <summary>
+        /// Finds a stable matching between two equal sets of elements (fills EngagedTo properties).
+        /// time complexity: O(n^2), where n - array size.
+        /// Guarantees:
+        /// - Everyone is matched
+        /// - Matches are stable (there is no better accepter, for any given proposer, which would accept a new match).
+        /// Presented and proven by David Gale and Lloyd Shapley in 1962.
+        /// </summary>
+        public static void Match(Proposer[] proposers, Accepter[] accepters)
+        {
+            if (proposers.Length != accepters.Length)
+            {
+                throw new ArgumentException("Collections must have equal count");
+            }
+
+            foreach (var proposer in proposers)
+            {
+                proposer.EngagedTo = null;
+            }
+
+            foreach (var accepter in accepters)
+            {
+                accepter.EngagedTo = null;
+            }
+
+            var matchingSession = proposers.Select(p => (Proposer: p, Accepters: accepters.OrderBy(a => p.Score(a)).ToList())).ToArray();
```

Comment:
There is this list already in   `public List<Accepter> PreferenceOrder { get; set; } = new List<Accepter>();`.
So just need to pass p.